packages/livekit-rtc/src/participant.ts

```diff
@@ -207,6 +223,222 @@ export class LocalParticipant extends Participant {
     });
   }
 
+  /**
+   * Returns a `StreamWriter` instance that allows to write individual chunks of text to a stream.
+   * Well suited for TTS and/or streaming LLM output.
+   */
+  async streamText(options?: {
+    topic?: string;
+    extensions?: Record<string, string>;
+    destinationIdentities?: Array<string>;
+  }): Promise<TextStreamWriter> {
+    const senderIdentity = this.identity;
+    const streamId = crypto.randomUUID();
+    const destinationIdentities = options?.destinationIdentities;
+
+    const headerReq = create(SendStreamHeaderRequestSchema, {
+      senderIdentity,
+      destinationIdentities,
+      localParticipantHandle: this.ffi_handle.handle,
+      header: {
+        streamId,
+        mimeType: 'text/plain',
+        topic: options?.topic ?? '',
+        timestamp: numberToBigInt(Date.now()),
+        extensions: options?.extensions,
+        contentHeader: {
+          case: 'textHeader',
+          value: {
+            operationType: DataStream_OperationType.CREATE,
+            version: 0,
+            replyToStreamId: '',
+            generated: false,
+          },
+        },
+      },
+    });
+
+    await this.sendStreamHeader(headerReq);
+
+    let chunkId = 0;
+    const localHandle = this.ffi_handle.handle;
+    const sendChunk = this.sendStreamChunk;
+
+    const writableStream = new WritableStream<[string, number?]>({
+      // Implement the sink
+      write([textChunk, overrideChunkId]) {
+        const textInBytes = new TextEncoder().encode(textChunk);
+
+        if (textInBytes.byteLength > STREAM_CHUNK_SIZE) {
+          this.abort?.();
+          throw new Error('chunk size too large');
+        }
+
+        return new Promise(async (resolve) => {
+          // FIXME we need an equivalent for this on the rust layer
+          // await localP.engine.waitForBufferStatusLow(DataPacket_Kind.RELIABLE);
```

Comment:
@theomonnom How does Rust currently handle the datachannel buffer? Do we have an event yet and/or a different mechanism for backpressure?